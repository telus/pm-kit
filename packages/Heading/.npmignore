*.stories.{js,mdx}
rollup.config.js
Heading.jsx
__tests__/