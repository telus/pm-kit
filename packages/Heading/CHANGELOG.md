# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0 (2020-01-22)

### Features

- **heading:** added Heading package ([#27](https://github.com/telus/pm-kit/issues/27)) ([7118026](https://github.com/telus/pm-kit/commit/711802630eed84107514ab40224483d776de8989))
