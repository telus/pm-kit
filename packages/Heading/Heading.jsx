import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import { css, jsx } from '@emotion/core'

const h1 = css`
  font-size: 3rem;
  font-weight: 500;
  line-height: 3rem;
  letter-spacing: -0.5px;
`

const h2 = css`
  font-size: 2rem;
  font-weight: 500;
  line-height: 3.25rem;
  letter-spacing: -0.5px;
`

const h3 = css`
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 1.8rem;
  letter-spacing: -0.5px;
`
const h4 = css`
  font-size: 1.125rem;
  font-weight: 500;
  line-height: 1.35rem;
  letter-spacing: -0.5px;
`

const Heading = ({ level, tag = level, children, forwardedRef, ...rest }) => {
  let variantStyles

  switch (level) {
    case 'h2':
      variantStyles = h2
      break
    case 'h3':
      variantStyles = h3
      break
    case 'h4':
      variantStyles = h4
      break
    default:
      variantStyles = h1
      break
  }

  return jsx(tag || 'h1', { css: variantStyles, ref: forwardedRef, ...rest }, children)
}

Heading.propTypes = {
  /**
   * The visual level of the heading. If `tag` is not specified, then `level` determines what HTML element to render.
   */
  level: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']).isRequired,
  /**
   * The semantic level of the heading. Renders the specified HTML element, otherwise it matches `level`.
   */
  tag: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'div', 'span']),
  /**
   * The content. Can be text, other components, or HTML elements.
   */
  children: PropTypes.node.isRequired,
}

Heading.defaultProps = {
  tag: undefined,
  level: 'h1',
}

const HeadingWithRef = forwardRef((props, ref) => <Heading {...props} forwardedRef={ref} />)

export default HeadingWithRef
