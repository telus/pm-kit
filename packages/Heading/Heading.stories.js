import React from 'react'
import { withKnobs, text, select, boolean } from '@storybook/addon-knobs'
import Heading from './Heading.jsx'
import { version } from './package.json'

export default {
  title: 'Design System|Heading',
  component: Heading,
  parameters: {
    componentSubtitle: `version ${version}`,
  },
  decorators: [withKnobs],
}

export const Heading1 = () => (
  <Heading tag="h1" level="h1">
    Heading 1
  </Heading>
)

export const Heading2 = () => (
  <Heading level="h2" tag="h2">
    Heading 2
  </Heading>
)

export const Heading3 = () => (
  <Heading level="h3" tag="h3">
    Heading 3
  </Heading>
)

export const Heading4 = () => (
  <Heading level="h4" tag="h4">
    Heading 4
  </Heading>
)

export const Playground = () => (
  <Heading
    level={select('Level', ['h1', 'h2', 'h3', 'h4'])}
    tag={select('Tag', [undefined, 'h1', 'h2', 'h3', 'h4', 'div', 'span'])}
    children={text('Text', 'Edit Text')}
  />
)

Playground.story = {
  name: 'playground',
  parameters: { docs: { page: null, disable: true } },
}
