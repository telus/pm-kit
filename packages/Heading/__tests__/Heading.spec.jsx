import React from 'react'
import { render } from 'enzyme'

import Heading from '../Heading'

describe('Heading', () => {
  it('renders', () => {
    const heading = render(<Heading />)

    expect(heading).toMatchSnapshot()
  })
})
