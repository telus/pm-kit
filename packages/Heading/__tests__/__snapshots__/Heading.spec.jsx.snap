// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders 1`] = `
.emotion-0 {
  font-size: 3rem;
  font-weight: 500;
  line-height: 3rem;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
}

<h1
  class="emotion-0"
>
  Test
</h1>
`;
